packages/graphiql/src/api/providers/GraphiQLSessionProvider.tsx

```diff
@@ -34,6 +36,7 @@ export const initialState: SessionState = {
   sessionId: 0,
   operation: { uri: 'graphql://graphiql/operations/0.graphql' },
   variables: { uri: 'graphql://graphiql/variables/0.graphql' },
+  headers: { uri: 'graphql://graphiql/variables/1.graphql' },
```

Comment:
I think this is what we want here? The problem would be that the session state would be trying to pull from whatever the second tab is regardless of whether the headers editor is enabled.